src/state/farms/fetchPublicFarmData.ts

```diff
@@ -0,0 +1,92 @@
+import BigNumber from 'bignumber.js'
+import { Address, Token } from 'config/constants/types'
+import masterchefABI from 'config/abi/masterchef.json'
+import erc20 from 'config/abi/erc20.json'
+import { getAddress, getMasterChefAddress } from 'utils/addressHelpers'
+import { BIG_TEN } from 'utils/bigNumber'
+import multicall from 'utils/multicall'
+
+const fetchFarm = async (pid: number, lpAddresses: Address, token: Token, quoteToken: Token) => {
```

Comment:
We have an issue with typing `FarmConfig`, `Farm` and the state in redux. To mitigate this I extracted the needed arguments for more accurate typing.

Reply:
I also removed the merging of the config to keep this function more pure.